### DISCUSSION TITLE
How do I make the wiki to go through the entire wiki folder including subfolders?

### DISCUSSION THREAD

**Discussion by blanquerna** [datenstrom/community#880] [Ask a question] [answered]
The option to number the pages is a good alternative to order content. Combined with numbered folders it is a fantastic option.
I have solved the problem of the menu (it only shows the first level) with a small recursive function that supports different levels of directories in the layout navigation.html.

It hasn't worked with the tags.

The page options [wikipages] and [wikitags] show only the folder pages and the main pages of the first level folders or the tags corresponding to these pages. If they ignore the pages and their tags contained in the subfolders.

The wiki (and I think the blog) does not act on the whole, but on the folders and only indexes / displays the folder.

When calling a tag (/tag=Agency/) it shows only the pages with this tag from the main folder and fails in any other case, even if it exists in the subfolder from where it was called.

The links in the tags use the local directory as a reference:
http://localhost/wiki/bloc09/tag=sostenibilitat/ and not the root of the wiki
http://localhost/wiki/tag=sostenibilitat/

I would like to be able to change the list of tags and have it go through the entire wiki folder (with subfolders) but I haven't located the API function that manages it: **What is the function that creates the list of Tags?**

I have created a simple page layout to show all the tags and/or authors with the correct links to the different pages. I would like to be able to replace the current pages of [tag=] and [tag=agency] with my page. **How could I redirect these pages to the new layout?**

I am testing and designing on the page:
https://wiki.cibame.eu/
where you can see how the menu works and the complete list of tags in the sidebar.

The page https://wiki.cibame.eu/wiki/etiquetes shows the layout with which I want to update the pages [tag=]

If you think it can be of general utility, tell me.

**Comment by annaesvensson** [comment]
Hello Blanquerna!

First of all: could you please edit the title and put your question there? For example "Can I make a wiki with subfolders?" or "How do I make the wiki to go through the entire wiki folder (with subfolders)?". Tip: You can use the `Edit` button at the top right corner. Otherwise it is not clear from the title what you are asking for and [the whole discussion risks being flagged and deleted](https://github.com/datenstrom/community/discussions/734).

**Reply by blanquerna** [reply]
Thanks for the observation. I write the question in the title.

**Comment by annaesvensson** [comment, marked as answer]
Thank your for the detailed information. The default behaviour of the [wiki extension](https://github.com/annaesvensson/yellow-wiki) is to use files in the current folder, it does not go through subfolders. You can change this behaviour by configuring a wiki start location in the settings. For example open file `system/extensions/yellow-system.ini` and change `WikiStartLocation: /wiki/`. 

Regarding your API question. I'm sure you already know that the main code is in file `system/extensions/wiki.php`. The functions responsible for creating a list of wiki tags are `getShortcutWikitags()` and `getWikiPages()`, the later one is probably the interesting one, it returns a collection of pages. This collection of pages decides what will be displayed, where tags are extracted from and so forth. If you configure a wiki start location in the settings, then the collection of pages contains wiki pages from anywhere on your website... and the rest should work automatically. Layout files are mentioned in the documentation, see [wiki settings](https://github.com/annaesvensson/yellow-wiki#settings).

Tip: You can show detected wiki pages, they are available on your website as `http://website/wiki/special:pages/`.   The URL is actually handled by `http://website/wiki/` and the remaining part, in this case `special:pages`, are location arguments for that page. You can use the file `system/layouts/wiki-start.html` for customising that page.

Hope this is what you are looking for.

**Comment by blanquerna** [comment]
Thanks for the quick reply.

I have gone through everything again and the problem was in the configuration:
I thought it was fine, but instead of ":" there was "=" and it obviously didn't work correctly.
Now when reviewing and correcting everything works fine.

Sorry for a stupid mistake.

I think that the problem of the labels is solved: they all appear and it addresses the pages correctly.

I think the menu is another problem because in the basic templates only one level is provided in the "navigation.html" layout. I have left in the template https://github.com/blanquerna/yellow-delta a multi-level menu model.

I plan to leave a page with reference to all tags, authors so that they do not appear on all pages and thus be able to eliminate the sidebar section.

The response page for "/tag=/" is a preconfigured page that I haven't found the layout of.

**How could I add at the end or at the beginning the list of all the Tags?**

**Comment by annaesvensson** [comment]
Let's stick to one question per discussion... you know the drill. 😄
